spec/mailboxes/reply_mailbox_spec.rb

```diff
@@ -59,12 +60,37 @@
         reply_mail_without_uuid.mail['In-Reply-To'] = '<conversation/6bdc3f4d-0bec-4515-a284-5d916fdde489/messages/123@test.com>'
       end
 
-      it 'find channel with reply to mail' do
+      it 'find channel with in-reply-to mail' do
         described_subject
         expect(conversation_1.messages.last.content).to eq("Let's talk about these images:")
       end
     end
 
+    context 'with reply_to email address present' do
+      let(:email_channel) { create(:channel_email, email: 'test@example.com', account: account) }
+      let(:conversation_1) do
+        create(
+          :conversation,
+          assignee: agent,
+          inbox: email_channel.inbox,
+          account: account,
+          additional_attributes: { mail_subject: "Discussion: Let's debate these attachments" }
+        )
+      end
+
+      before do
+        conversation_1.update!(uuid: '6bdc3f4d-0bec-4515-a284-5d916fdde489')
+      end
+
+      it 'prefer reply-to over from address' do
+        described_class.receive reply_to_mail
+        expect(conversation_1.messages.last.content).to eq("Let's talk about these images:")
+
+        email = conversation_1.messages.last.content_attributes['email']
+        expect(reply_to_mail.mail['Reply-To'].value).to include(email['from'][0])
```

Comment:
> 2. When a new message is created in an existing conversation, then REPLY-TO would be given preference to select the contact. 

When a new message is created in the already present conversation with the message-id/in-reply-to/to address we just append the new message we don't append the contact anywhere.

Should I make that change?

The above test checks, the mail body is saved in content attributes with a reply-to address as from address.

Reply:
Let us do that as a separate PR. I guess we are not creating a new contact when a new message comes in.